stats/datadog/collector.go

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datadog
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/loadimpact/k6/stats/statsd/common"
+	log "github.com/sirupsen/logrus"
+	null "gopkg.in/guregu/null.v3"
+)
+
+type tagHandler sort.StringSlice
+
+func (t tagHandler) filterTags(tags map[string]string, group string) []string {
+	var res []string
+
+	for key, value := range tags {
+		if value != "" && t.contains(key) {
+			res = append(res, key+":"+value)
```

Comment:
Not certain how well this will hold up if I have a `:` in my tag name (after all, I can define custom tags) or value...

Reply:
hm ... maybe we should restrict tags as we have restricted metrics ? https://github.com/loadimpact/k6/blob/master/js/modules/k6/metrics/metrics.go#L35 . We could escape the ':' somehow but in my opinion this will not help much ... maybe just leave it and document that having `:` in the tags will produce not desired tags in datadog ? after all it won't break just if you have tag with name `something:else` and value `myvalye` you will get tag with key `something` and value ``else:mayvalue` in datadog